src/main/java/io/supertokens/webserver/api/totp/VerifyTotpAPI.java

```diff
@@ -65,10 +65,6 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
 
                 StorageAndUserIdMapping storageAndUserIdMapping = getStorageAndUserIdMappingForTenantSpecificApi(
                         req, userId, UserIdType.ANY);
-
-                if (storageAndUserIdMapping.userIdMapping != null) {
-                    userId = storageAndUserIdMapping.userIdMapping.superTokensUserId;
-                }
                 storage = storageAndUserIdMapping.storage;
             } catch (UnknownUserIdException e) {
```

Comment:
bad pattern

Reply:
You just have to do storage = getTenantStorage(req);

Reply:
Done